gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -148,18 +159,15 @@ private void setLimiterReportKeyListToWorkUnits(List<WorkUnit> workUnits, List<S
     isFullExtract = state.getPropAsBoolean(ConfigurationKeys.EXTRACT_IS_FULL_KEY);
 
     try {
-      this.kafkaConsumerClient =
-          kafkaConsumerClientResolver
-          .resolveClass(
-              state.getProp(GOBBLIN_KAFKA_CONSUMER_CLIENT_FACTORY_CLASS,
-                  DEFAULT_GOBBLIN_KAFKA_CONSUMER_CLIENT_FACTORY_CLASS)).newInstance()
-          .create(ConfigUtils.propertiesToConfig(state.getProperties()));
+      this.kafkaConsumerClient = kafkaConsumerClientResolver.resolveClass(state
+          .getProp(GOBBLIN_KAFKA_CONSUMER_CLIENT_FACTORY_CLASS, DEFAULT_GOBBLIN_KAFKA_CONSUMER_CLIENT_FACTORY_CLASS))
+          .newInstance().create(ConfigUtils.propertiesToConfig(state.getProperties()));
 
       List<KafkaTopic> topics = getFilteredTopics(state);
+      this.topicsToProcess = new HashSet<>(topics.stream().map(KafkaTopic::getName).collect(toList()));
```

Comment:
you can do `topics.stream().map(KafkaTopic::getName).collect(Collectors.toSet)` instead